src/snapshots/RandomAccessIterator.cpp

```diff
@@ -9,6 +9,30 @@ inline RandomAccessIterator<iterator_t>::RandomAccessIterator(iterator_t it)
 template <typename iterator_t>
 RandomAccessIterator<iterator_t>::~RandomAccessIterator() = default;
 
+template <typename iterator_t>
+#ifdef __HIPCC__
```

Comment:
Ok, let us add a comment on each of these work-around lines in which ROCm versions we saw this and then patch it in for now.

On each `ifdef` I would add:
```suggestion
#ifdef __HIPCC__  // ROCm 6.2.4 issue, see #1797
```

That way, we can remove it in a few years down with confidence again.